File: app/src/main/java/ds/vkplus/exception/VKException.java
package ds.vkplus.exception;

import ds.vkplus.network.model.ApiResponse;

public class VKException extends RuntimeException {

	public static final int CODE_UNKNOWN = -1;
	public static final int CODE_TOKEN_OBSOLETED = 5;

	public int code;
	public String message;


	public VKException(ApiResponse.Error e) {
		code = e.error_code;
		message = e.error_msg;
	}


	public VKException() {
		code = CODE_UNKNOWN;
		message = "Unknown Error";

	}
}

File: app/src/main/java/ds/vkplus/exception/VKException.kt
package ds.vkplus.exception

import ds.vkplus.model.ApiResponse

class VKException : RuntimeException {

    var code: Int = 0
    var mess: String


    constructor(e: ApiResponse.Error) {
        code = e.error_code
        mess = e.error_msg
    }


    constructor() {
        code = CODE_UNKNOWN
        mess = "Unknown Error"

    }

    companion object {
        const val CODE_UNKNOWN: Int = -1
        const val CODE_NETWORK: Int = -2
        const val CODE_TOKEN_OBSOLETED: Int = 5
    }
}
